govcd/edgegateway.go

```diff
@@ -1167,3 +1167,131 @@ func validateUpdateLBGeneralParams(logLevel string) error {
 
 	return nil
 }
+
+// getVnics retrieves a structure of type EdgeGatewayVnics which contains network interfaces
+// available in Edge Gateway
+func (egw *EdgeGateway) getVnics() (*types.EdgeGatewayVnics, error) {
+	if !egw.HasAdvancedNetworking() {
+		return nil, fmt.Errorf("only advanced edge gateway supports vNics")
+	}
+
+	httpPath, err := egw.buildProxiedEdgeEndpointURL(types.EdgeVnicConfig)
+	if err != nil {
+		return nil, fmt.Errorf("could not get Edge Gateway API endpoint: %s", err)
+	}
+
+	vnicConfig := &types.EdgeGatewayVnics{}
+	_, err = egw.client.ExecuteRequest(httpPath, http.MethodGet, types.AnyXMLMime,
+		"unable to edge gateway vnic configuration: %s", nil, vnicConfig)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return vnicConfig, nil
+}
+
+// GetVnicIndexFromNetworkNameType returns *int of vNic index for specified network name and network type
+// networkType one of: 'internal', 'uplink', 'trunk', 'subinterface'
+// networkName cannot be empty
+func (egw *EdgeGateway) GetVnicIndexFromNetworkNameType(networkName, networkType string) (*int, error) {
```

Comment:
I suggest naming these new methods with "By" instead "From" and with an "And", i.e.: GetVnicIndex**By**NetworkName**And**Type

This would make these names consistent/similar to methods like:
GetCatalog**By**Name**Or**Id

CC: @dataclouder

Reply:
Fixed